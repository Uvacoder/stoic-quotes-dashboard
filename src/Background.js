import React from 'react';
import styled from 'styled-components';

import Image from './Image';
import getRandomIntInclusive from './getRandomIntInclusive';

const Container = styled.div`
  width: 100%;
  height: 100%;
  margin: 0;
  position: absolute;
  z-index: auto;
`;

function Background() {
  const [photo, setPhoto] = React.useState();

  React.useEffect(() => {
    const randomInt = getRandomIntInclusive(1, 100)
    const getRandomPhoto = () => {
      const photo = (`https://source.unsplash.com/collection/2281806/${randomInt}`)
      setPhoto(photo);
    };
    getRandomPhoto();
  }, []);
  return (
    <Container>
      <Image src={photo} />
    </Container>
  );
}

export default Background;