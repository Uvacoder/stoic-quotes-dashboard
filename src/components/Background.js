import React, { useState, useEffect } from 'react'
import styled from 'styled-components'
import Image from './Image'
import CenterContainer from './shared-styles/CenterContainer'

const Container = styled(CenterContainer)`
  width: 100%;
  height: 100%;
  margin: 0;
  position: absolute;
  object-position: 50% 50%;
  z-index: auto;
  background-color: black;
`

const Background = () => {
  const [photo, setPhoto] = useState()

  useEffect(() => {
    const url = 'https://source.unsplash.com/collection/477172/1600x900/daily'
    const getRandomPhoto = () => {
      setPhoto(url)
    }
    getRandomPhoto()
  }, [])

  return (
    <Container>
      <Image src={photo} />
    </Container>
  )
}

export default Background
