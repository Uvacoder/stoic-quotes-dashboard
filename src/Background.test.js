import React from 'react';
import renderer from 'react-test-renderer';
import Background from './Background'

describe('<Background />', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Background />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})