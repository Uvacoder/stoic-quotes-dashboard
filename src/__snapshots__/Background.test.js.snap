// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> snapshot renders 1`] = `
<div
  className="sc-bwzfXH kGrqoq"
>
  <img
    className="sc-bdVaJa inUwYz"
  />
</div>
`;
